Sources/Apollo/RequestCreator.swift

```diff
@@ -50,48 +74,34 @@ public struct RequestCreator {
     } else {
       formData = MultipartFormData()
     }
-    
-    // Make sure all fields for files are set to null, or the server won't look
-    // for the files in the rest of the form data
-    let fieldsForFiles = Set(files.map { $0.fieldName })
-    var fields = requestBody(for: operation, sendOperationIdentifiers: sendOperationIdentifiers)
-    var variables = fields["variables"] as? GraphQLMap ?? GraphQLMap()
-    for fieldName in fieldsForFiles {
-      if
-        let value = variables[fieldName],
-        let arrayValue = value as? [JSONEncodable] {
-        let updatedArray: [JSONEncodable?] = arrayValue.map { _ in nil }
-          variables.updateValue(updatedArray, forKey: fieldName)
+
+    let fields = requestBody(for: operation)
+    for (name, data) in fields {
+      if let data = data as? GraphQLMap {
+        let data = try serializationFormat.serialize(value: data)
+        formData.appendPart(data: data, name: name)
+      } else if let data = data as? String {
+        try formData.appendPart(string: data, name: name)
       } else {
-        variables.updateValue(nil, forKey: fieldName)
-      }
-    }
-    fields["variables"] = variables
-    
-    let operationData = try serializationFormat.serialize(value: fields)
-    formData.appendPart(data: operationData, name: "operations")
-    
-    var map = [String: [String]]()
-    if files.count == 1 {
-      let firstFile = files.first!
-      map["0"] = ["variables.\(firstFile.fieldName)"]
-    } else {
-      for (index, file) in files.enumerated() {
-        map["\(index)"] = ["variables.\(file.fieldName).\(index)"]
+        try formData.appendPart(string: data.debugDescription, name: name)
       }
     }
-    
-    let mapData = try serializationFormat.serialize(value: map)
-    formData.appendPart(data: mapData, name: "map")
-    
-    for (index, file) in files.enumerated() {
-      formData.appendPart(inputStream: file.inputStream,
-                          contentLength: file.contentLength,
-                          name: "\(index)",
-                          contentType: file.mimeType,
-                          filename: file.originalName)
+
+    files.forEach {
+      formData.appendPart(inputStream: $0.inputStream, contentLength: $0.contentLength, name: $0.fieldName, contentType: $0.mimeType, filename: $0.originalName)
     }
-    
+
     return formData
   }
 }
+
+// Helper struct to create requests independently of HTTP operations.
+struct ApolloRequestCreator: RequestCreator {
+
+  /// Default Apollo request creator
+  ///
+  /// - Returns: Default apollo request creator
+  static func `default`() -> RequestCreator {
```

Comment:
does this need to exist? Wouldn't you just be able to create the default one using `ApolloRequestCreator()`?